src/compiler/chainrules.jl

```diff
@@ -0,0 +1,128 @@
+const chainrules_fallback = which(rrule, Tuple{Any})
+
+"""
+  has_chain_rrule(T)
+
+For a type-tuple `T` e.g. `Tuple{typeof(f), Int, Float64}`, checks if there is a `rrule` defined for it.
+Excluding the generic fallback.
+The first return value is `true` if the `rrule` exists, `false` otherwise.
+If it does not, then the second argument is a list of edges to attach to the CodeInfo for a generated function,
+such that if a suitable rule is defined later, the generated function will recompile.
+"""
+function has_chain_rrule(T)
+  m = meta(Tuple{typeof(rrule),T.parameters...})
+  if m.method !== chainrules_fallback
+    # found a rrule, no need to add any edges
+    return true, nothing
+  end
+
+  # Could be a kwarg function, handle that case
+  if is_kwfunc(T.parameters...)
+    # Need to check for rrule for function not the kwfunction.
+    base_T = Tuple{T.parameters[3:end]...}
+    return has_chain_rrule(base_T)
+  end
+
+  # did not find anything, will have to attach edges so it recompiles if one is added
+  @assert m.code.edges !== nothing
+  return false, m.code.edges
```

Comment:
This is not going to work preJulia-1.3
I guess we can just return `[]` in that case and not support the loading of `rrules` after the fact (without `Zygote.refresh()`)

Will have to add that condition here and in `_pullback`